packages/aws-cdk-lib/core/lib/resolvable.ts

```diff
@@ -158,6 +158,8 @@ export class DefaultTokenResolver implements ITokenResolver {
       // The token might have returned more values that need resolving, recurse
       resolved = context.resolve(resolved);
       resolved = postProcessor.postProcess(resolved, context);
+      // resolve again since postProcess might have added more tokens (e.g. overriding)
+      resolved = context.resolve(resolved);
```

Comment:
It looks like we are passing the `context` through to the `postProcess`, but it
isn't making its way all the way through to the `processor`. Is there a way we
could pass it all the way through and use that to replace the
`Tokenization.resolve` function?

Reply:
Thanks, I'll check that way

Reply:
@corymhall 
I investigated it, and please correct me if my understanding is wrong 🙏 

If we take that approach, we can use `context.resolve` here instead of `Tokenization.resolve`, and it can also settle the issue.

https://github.com/aws/aws-cdk/blob/04323c44da6b8821a361a36bb2feadde77d54c35/packages/aws-cdk-lib/core/lib/cfn-resource.ts#L441-L447

The disadvantage is that, to pass the context, we have to change the signature of `_toCloudFormation` function something like `_toCloudFormation(context?)`. It requires changes in many files and I don't think it is a preferred approach. 

Isn't it more reasonable to resolve tokens outside of the postProcess function?

Reply:
I was thinking of adding it to the `PostResolveToken` `processor` method as an optional property

https://github.com/aws/aws-cdk/blob/b1e649a24375ca13a96c1b667ef27e92577b329f/packages/aws-cdk-lib/core/lib/util.ts#L82.

So then it would be something like
```ts
[this.logicalId]: new PostResolveToken({
            ...
          }, (resourceDef, context) => {
```

Reply:
I like Cory's solution. `PostResolveToken` is private and so is free to have its API changed. This will (hopefully?) cleanly achieve the desired solution.

Reply:
@corymhall @rix0rrr 
Thanks, I think I have to study about CDK tokens more 😄  Will try the approach anyway.

One question is, why not resolve tokens after the deepMerge? (i.e. my original approach)
It can remove some hacks currently implemented like `removeEmpty: false` or `MERGE_EXCLUDE_KEYS`, and simplify the existing logic.

Is it performance-wise reason (to reduce recursion), or is there any edge case not to work properly that I am not aware of?